### DISCUSSION TITLE
Is MomentumCombine the inverse of ExpandScalarProduct?

### DISCUSSION THREAD

**Discussion by ElJefeDelDesierto** [FeynCalc/feyncalc#271] [Q&A] [answered]
Hello FeynCalc team!

First, I would like to thank you for your fantastic work and excellent support. My question (perhaps naive) is the following: The documentation says that $\text{MomentumCombine}$ is the inverse operation of $\text{MomentumExpand}$ and $\text{ExpandScalarProduct}$; if I calculate $SPD[p-k,p-k]//\text{ExpandScalarProduct}$ I get $SPD[k,k]-2SPD[k,p]+SPD[p,p]$; now, what I want to do is the inverse operation, that is, start from $SPD[k,k]-2SPD[k,p]+SPD[p,p]$ and get $SPD[p-k,p-k]$. According to the documentation, I should use $\text{MomentumCombine}$, but when I use it, what I get is $SPD[k,k-2p]+SPD[p,p]$, not $SPD[p-k,p-k]$ (even if applied twice, it gives the same result). How is it possible to get $SPD[p-k,p-k]$ starting from $SPD[k,k]-2SPD[k,p]+SPD[p,p]$?

Of course I could do it by hand, but my expressions are so huge that doing it this way is impossible.
Thanks in advance.

**Comment by vsht** [comment, marked as answer]
Hi,

MomenumCombine can handle some simple cases, like $p.a + p.b = p.(a+b)$ but not squares and similar stuff. I also don't know an efficient way to undo such expansions in general, except for doing it by brute force and trying out all possible combinations.

Am 24. Mai 2024 18:24:03 UTC schrieb ElJefeDelDesierto ***@***.***>:
